cmd/dashboard.go

```diff
@@ -76,19 +83,81 @@ var dashboardCmd = &cobra.Command{
 }
 
 func runPortforward(podName string, localPort int) error {
-	cmd, err := exec.LookPath("kubectl")
+	stopChannel := make(chan struct{}, 1)
+	readyChannel := make(chan struct{})
+
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, os.Interrupt)
+	defer signal.Stop(signals)
+
+	go func() {
+		<-signals
+		if stopChannel != nil {
+			close(stopChannel)
+		}
+	}()
+
+	// Open the Dashboard in a browser when the port-forward is established
+	go func() {
+		<-readyChannel
+		openInBrowser(fmt.Sprintf("http://localhost:%d", localPort))
+	}()
+
+	fw, err := newPortforwarder(podName, localPort, stopChannel, readyChannel)
 	if err != nil {
 		return err
 	}
-	args := []string{"kubectl", "--namespace", ingressNamespace, "port-forward", podName, fmt.Sprintf("%d:80", localPort)}
+	return fw.ForwardPorts()
+}
 
-	env := os.Environ()
+func newPortforwarder(podName string, localPort int, stopChannel, readyChannel chan struct{}) (*portforward.PortForwarder, error) {
+	config, err := restClientConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	url, err := portforwardReqURL(config, podName)
+	if err != nil {
+		return nil, err
+	}
+
+	dialer, err := remotecommand.NewExecutor(config, "POST", url)
+	if err != nil {
+		return nil, err
+	}
+
+	ports := []string{fmt.Sprintf("%d:80", localPort)}
```

Comment:
shouldn't be `80` obtained somehow?

Reply:
Maybe, but given we know for sure it's going to be port 80, I think we can avoid the extra API call.

Reply:
well, other people (like me) may see it that obvious. For example you also know for sure what URL the port forward will use but anyway you are obtaining it just in case it changes (and for the sake of readability)

Reply:
I would argue the difference between building the port-forward request URL is that we control the ingress controller manifests. We don't control the Kubernetes API though, so that could easily change from version to version.

Reply:
I think Andres is just suggesting a comment, which seems reasonable.  We don't want a future reader to have to work this out for themselves.

Reply:
yeah, I just didn't know where the `80` came from

Reply:
sorry I misunderstood, I thought you meant parsing the Service object to get the port. Will add a comment.

Reply:
yes, that's what I meant, but as far as it is clear where it comes from I am fine